scripts/verify-checksum-models.py

```diff
@@ -0,0 +1,64 @@
+import os
+import hashlib
+
+# Define the path to the llama directory (parent folder of script directory)
+llama_path = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir))
+
+# Define the file with the list of hashes and filenames
+hash_list_file = os.path.join(llama_path, "SHA256SUMS")
+
+# Check if the hash list file exists
+if not os.path.exists(hash_list_file):
+    print(f"Hash list file not found: {hash_list_file}")
+    exit(1)
+
+# Read the hash file content and split it into an array of lines
+with open(hash_list_file, "r") as f:
+    hash_list = f.read().splitlines()
+
+# Create an array to store the results
+results = []
+
+# Loop over each line in the hash list
+for line in hash_list:
+    # Split the line into hash and filename
+    hash_value, filename = line.split("  ")
+
+    # Get the full path of the file by joining the models path and the filename
+    file_path = os.path.join(llama_path, filename)
+
+    # Informing user of the progress of the integrity check
+    print(f"Verifying the checksum of {file_path}")
+
+    # Check if the file exists
+    if os.path.exists(file_path):
+        # Calculate the SHA256 checksum of the file using hashlib
+        with open(file_path, "rb") as f:
+            file_hash = hashlib.sha256(f.read()).hexdigest()

```

Comment:
`f.read()` reads the whole file into the memory - this will behave poorly for big files that do not fit in RAM

you want to do something akin to:

``` python
def sha256sum(filename):
    BLOCKSIZE = 16 * 1024 * 1024
    h  = hashlib.sha256()
    b  = bytearray(BLOCKSIZE)
    mv = memoryview(b)
    with open(filename, 'rb', buffering=0) as f:
        n = f.readinto(mv)
        if n < BLOCKSIZE:
            h.update(mv[:n])
            break
        else:
            h.update(mv)
    return h.hexdigest()
```

Reply:
Thank you for the suggestion, I've updated the script.